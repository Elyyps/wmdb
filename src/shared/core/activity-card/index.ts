export * from './activity-card.component'