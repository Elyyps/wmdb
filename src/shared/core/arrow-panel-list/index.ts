export * from "./arrow-panel-list.component";
