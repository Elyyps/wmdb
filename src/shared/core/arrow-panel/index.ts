export * from "./arrow-panel.component";
