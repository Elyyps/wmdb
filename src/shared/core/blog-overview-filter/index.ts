export * from "./blog-overview-filter.component";
