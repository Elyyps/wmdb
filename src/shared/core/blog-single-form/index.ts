export * from "./blog-single-form.component";
