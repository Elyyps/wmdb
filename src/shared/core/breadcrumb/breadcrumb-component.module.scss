.breadcrumb {
  margin: 0;
  padding: 16px 0 ;
  list-style-type: none;
  display: flex;
  align-items: center;
  font-size: 14px;
  margin: 0 -12px;
  >li{
    position: relative;
    >a, >span{
      display: flex;
      align-items: center;
      color: $global-color;
      text-decoration: none;
      padding: 0 10px;
      svg{
        margin-left: 10px;
        path{
          stroke:$gray-light;
          fill: none;
        }
      }
    }
    &:last-child{
      span{
        color: $gray-light;
      }
    }
  }
}
