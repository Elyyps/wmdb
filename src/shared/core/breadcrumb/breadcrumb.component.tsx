import * as React from "react";
import styles from "./breadcrumb-component.module.scss";
import { Link } from "react-router-dom";
import { IconComponent } from "../icon";
import ArrowRight from "@assets/icons/chevron-right.svg";
import { ILink } from "@app/api/core/link";

export interface IBreadcrumbProps {
  breadcrumbs: ILink[];
}

const Breadcrumb = ({ breadcrumbs }: IBreadcrumbProps) => (
  <React.Fragment>
    <ul className={styles["breadcrumb"]}>
      {breadcrumbs.map((item: ILink, key: number) =>
        item.url ? (
          <li key={key}>
            <Link to={item.url}>
              <span>{item.title}</span>
              <IconComponent icon={ArrowRight} size={"8px"} />
            </Link>
          </li>
        ) : (
          <li key={key}>
            <span>{item.title}</span>
          </li>
        )
      )}
    </ul>
  </React.Fragment>
);

export { Breadcrumb };
