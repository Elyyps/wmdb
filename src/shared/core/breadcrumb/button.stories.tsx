import React from "react";
import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { Breadcrumb } from "@app/core/breadcrumb";

storiesOf("Breadcrumb", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <Breadcrumb
      breadcrumbs={[
        { title: "Home", url: "/" },
        { title: "Another page", url: "/" },
        { title: "Current page", url: "" }
      ]}
    />
  ));
