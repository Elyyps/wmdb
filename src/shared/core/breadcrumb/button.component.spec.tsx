import React from "react";
import { shallow } from "enzyme";
import { Breadcrumb } from "@app/core/breadcrumb";

describe("[Button]", () => {
  it("should render without crashing", () => {
    const renderedComponent = shallow(
      <Breadcrumb
        breadcrumbs={[
          { title: "Home", url: "/" },
          { title: "Another page", url: "/" },
          { title: "Current page", url: "" }
        ]}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it("should render without crashing", () => {
    const renderedComponent = shallow(<Breadcrumb breadcrumbs={[]} />);
    expect(renderedComponent).toMatchSnapshot();
  });
  it("should render without crashing", () => {
    const renderedComponent = shallow(<Breadcrumb breadcrumbs={[{ title: "Home", url: "/" }]} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
