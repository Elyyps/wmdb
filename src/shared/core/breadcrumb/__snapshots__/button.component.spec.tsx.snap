// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button] should render without crashing 1`] = `
<Fragment>
  <div
    className="uk-container"
  >
    <ul
      className="breadcrumb"
    >
      <li
        key="0"
      >
        <Link
          to="/"
        >
          <span>
            Home
          </span>
          <IconComponent
            icon="chevron-right.svg"
            size="8px"
          />
        </Link>
      </li>
      <li
        key="1"
      >
        <Link
          to="/"
        >
          <span>
            Another page
          </span>
          <IconComponent
            icon="chevron-right.svg"
            size="8px"
          />
        </Link>
      </li>
      <li
        key="2"
      >
        <span>
          Current page
        </span>
      </li>
    </ul>
  </div>
</Fragment>
`;

exports[`[Button] should render without crashing 2`] = `
<Fragment>
  <div
    className="uk-container"
  >
    <ul
      className="breadcrumb"
    />
  </div>
</Fragment>
`;

exports[`[Button] should render without crashing 3`] = `
<Fragment>
  <div
    className="uk-container"
  >
    <ul
      className="breadcrumb"
    >
      <li
        key="0"
      >
        <Link
          to="/"
        >
          <span>
            Home
          </span>
          <IconComponent
            icon="chevron-right.svg"
            size="8px"
          />
        </Link>
      </li>
    </ul>
  </div>
</Fragment>
`;
